### PR TITLE
Fix bugprone-narrowing-conversions

### PR DESCRIPTION
**Pull request by sorru94**
Complete some tasks of: #112

### CODE REVIEW

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/astarte_device.c [hunk 1]
```diff
@@ -499,18 +499,23 @@ astarte_err_t astarte_device_stop(astarte_device_handle_t device)
...
 {
-    int len = 0;
+    size_t len = 0;
     const void *data = astarte_bson_serializer_get_document(bs, &len);
     if (!data) {
         ESP_LOGE(TAG, "Error during BSON serialization");
         return ASTARTE_ERR;
     }
+    if ((int) len < 0) {
+        ESP_LOGE(TAG, "BSON document is too long for MQTT publish.");
```
[the remark this answers is not in the document]
Comment (sorru94) [reply src/astarte_device.c:509]: Yep, that makes sense.

File: src/astarte_bson_serializer.c [hunk 2]
```diff
@@ -157,17 +162,17 @@ void astarte_bson_serializer_append_end_of_document(struct astarte_bson_serializ
 {
     astarte_byte_array_append_byte(&bs->ba, '\0');
 
-    uint8_t *sizeBuf = NULL;
-    INT32_TO_BYTES(bs->ba.size, sizeBuf)
+    uint8_t sizeBuf[4];
+    uint32_to_bytes(bs->ba.size, sizeBuf);
```
Comment (harlem88) [inline src/astarte_bson_serializer.c:166]: Is it right use `uint32_to_bytes` instead of `int32_to_bytes`?

File: src/astarte_bson_serializer.c [hunk 3]
```diff
@@ -188,19 +193,19 @@ void astarte_bson_serializer_append_int32(
...
 }
 
 void astarte_bson_serializer_append_binary(
-    struct astarte_bson_serializer_t *bs, const char *name, const void *value, int size)
+    struct astarte_bson_serializer_t *bs, const char *name, const void *value, size_t size)
 {
-    uint8_t *lenBuf = NULL;
-    INT32_TO_BYTES(size, lenBuf)
+    uint8_t lenBuf[4];
+    uint32_to_bytes(size, lenBuf);
```
Comment (harlem88) [inline src/astarte_bson_serializer.c:208]: same here

File: src/astarte_bson_serializer.c [hunk 4]
```diff
@@ -212,10 +217,10 @@ void astarte_bson_serializer_append_binary(
 void astarte_bson_serializer_append_string(
     struct astarte_bson_serializer_t *bs, const char *name, const char *string)
 {
-    int string_len = strlen(string);
+    size_t string_len = strlen(string);
 
-    uint8_t *lenBuf = NULL;
-    INT32_TO_BYTES(string_len + 1, lenBuf)
+    uint8_t lenBuf[4];
+    uint32_to_bytes(string_len + 1, lenBuf);
```
Comment (harlem88) [inline src/astarte_bson_serializer.c:223]: same here

File: src/astarte_bson_serializer.c (hunk 2 above)
Comment (sorru94) [reply src/astarte_bson_serializer.c:166]: It makes more sense to use an unsigned number to store a size as it should never be negative.

**Review by harlem88 — APPROVED** [review]

**Pull request merged**